AerofoilMac/AerofoilMac/AerofoilApplication.m

```diff
@@ -0,0 +1,13 @@
+#import "AerofoilApplication.h"
+#include "SDL.h"
+
+extern int SDL_SendQuit(void);
+
+@implementation AerofoilApplication
+
+- (void)terminate:(id)sender {
+	// TODO: Use Aerofoil method instead of private SDL method
```

Comment:
Hmm, the issue I was trying to describe here was that `SDL_SendQuit` doesn't seem to be a part of SDL's public headers, so as a workaround I forward declared the function myself and called it, which IMO is a bit of a hack…

The reason why I'm calling this explicitly to begin with: Normally SDL would call `SDL_SendQuit` if I didn't have this subclass of `NSApplication`, but the subclass exists to prevent SDL forcefully activating the app (this isn't consistent with normal Mac apps), and to allow adding/handling the native menu items. 

When I first wrote this I did try to add a quit event to the VOS queue rather than going through SDL, but I thought there wasn't a way to get a reference to the queue. But trying again I came up with this:
```
- (void)terminate:(id)sender {
	IGpDisplayDriver *displayDriver = PLDrivers::GetDisplayDriver();
	const GpDisplayDriverProperties& properties = displayDriver->GetProperties();
	if (GpVOSEvent *evt = properties.m_eventQueue->QueueEvent())
		evt->m_eventType = GpVOSEventTypes::kQuit;
}
```
Is that also fine?

Reply:
> Hmm, the issue I was trying to describe here was that `SDL_SendQuit` doesn't seem to be a part of SDL's public headers, so as a workaround I forward declared the function myself and called it, which IMO is a bit of a hack…

Oh.  You can use SDL_PushEvent to insert a quit event.

> I'm guessing this is also okay based on what you suggested for the about/preferences functions? :)

If you need global access to the event queue in the shell, you should use this instead:

`GpAppInterface_Get()->PL_GetDriverCollection()->GetDriver<GpDriverIDs::kEventQueue>();`